string switch
string action
---
int64 response